nuxeo-retention/src/main/java/org/nuxeo/retention/actions/ProcessRetentionEventAction.java

```diff
@@ -108,7 +118,10 @@ protected void scheduleInputEventBasedRule(RetentionRule rule) {
                  .append(" = '" + rule.getDocument().getId() + "'");
             for (String repositoryName : repositoryService.getRepositoryNames()) {
                 BulkCommand command = new BulkCommand.Builder(EvalInputEventBasedRuleAction.ACTION_NAME,
-                        query.toString(), SecurityConstants.SYSTEM_USERNAME).repository(repositoryName).build();
+                        query.toString(), SecurityConstants.SYSTEM_USERNAME).param(ACTION_EVENT_ID_PARAM, eventId)
+                                                                            .param(ACTION_EVENT_INPUT_PARAM, eventInput)
+                                                                            .repository(repositoryName)
+                                                                            .build();
                 bulkService.submit(command);
```

Comment:
a bulk command that is submitting another bulk command is dangerous, it hard to make it idempotent, retrying or replaying stream will create duplicates

Reply:
Maybe we can remove the intermediate ProcessRetentionEventAction.java. Instead, we will submit the `EvalInputEventBasedRuleAction` with a query such as 
```
"SELECT * FROM Document WHERE ecm:mixinType = 'Record AND ecm:isRecord = 1 AND ecm:retainUntil = TIMESTAMP 9999-01-01T00:00Z AND record:ruleIds IN (${rulesIds})"
```
where `ruleIds` are all the retention rules ids plugged on the given event retrieved in `RetentionBusinessEventListener.java`?

This can work and scale only if we don't have too many such rules which functionally should be the case.

Reply:
done in last commit